src/Domain/CoordinateMaps/SphericalTorus.cpp

```diff
@@ -74,14 +83,26 @@ SphericalTorus::jacobian(const std::array<T, 3>& source_coords) const {
   auto jacobian = make_with_value<tnsr::Ij<UnwrappedT, 3, Frame::NoFrame>>(
       dereference_wrapper(source_coords[0]), 0.0);
 
-  const auto r = radius(source_coords[0]);
-  const auto theta = M_PI * fraction_of_torus_ * source_coords[2];
-  const auto phi = pi_over_2_minus_theta_min_ * source_coords[1];
-  const UnwrappedT cos_theta = cos(theta);
-  const UnwrappedT sin_theta = sin(theta);
-  const UnwrappedT cos_phi = cos(phi);
-  const UnwrappedT sin_phi = sin(phi);
+  // In order to reduce number of memory allocations we use some slots of
+  // jacobian for storing temp variables as below.
 
+  auto& sin_theta = get<1, 1>(jacobian);
+  auto& cos_theta = get<1, 2>(jacobian);
+  get<2, 2>(jacobian) = M_PI * fraction_of_torus_ * source_coords[2];
+  cos_theta = cos(get<2, 2>(jacobian));
+  sin_theta = sin(get<2, 2>(jacobian));
+
+  auto& sin_phi = get<2, 0>(jacobian);
+  auto& cos_phi = get<2, 1>(jacobian);
+  get<2, 2>(jacobian) = pi_over_2_minus_theta_min_ * source_coords[1];
+  cos_phi = cos(get<2, 2>(jacobian));
+  sin_phi = sin(get<2, 2>(jacobian));
+
+  get<2, 2>(jacobian) = radius(source_coords[0]);
```

Comment:
This still does an allocation & copy. I think radius should take by `gsl::not_null`. `radius_inverse` is fine since we only use that on `double` I think

Reply:
Added an overload of `radius` using `not_null`